internal/sql/repository/CiArtifactRepository.go

```diff
@@ -109,35 +114,54 @@ func (impl CiArtifactRepositoryImpl) GetByWfId(wfId int) (*CiArtifact, error) {
 }
 
 // this method takes CD Pipeline id and Returns List of Artifacts Latest By last deployed
-func (impl CiArtifactRepositoryImpl) GetArtifactsByCDPipeline(cdPipelineId, limit int) ([]CiArtifact, error) {
+func (impl CiArtifactRepositoryImpl) GetArtifactsByCDPipeline(cdPipelineId, limit int, parentId int, parentType bean.WorkflowType) ([]CiArtifact, error) {
 	var artifactsA []CiArtifact
 	var artifactsAB []CiArtifact
-
-	queryFetchArtifacts := ""
-	queryFetchArtifacts = "SELECT cia.id, cia.data_source, cia.image, cia.image_digest, cia.scan_enabled, cia.scanned FROM ci_artifact cia" +
-		" INNER JOIN ci_pipeline cp on cp.id=cia.pipeline_id" +
-		" INNER JOIN pipeline p on p.ci_pipeline_id = cp.id" +
-		" WHERE p.id= ? ORDER BY cia.id DESC"
-	_, err := impl.dbConnection.Query(&artifactsA, queryFetchArtifacts, cdPipelineId)
-	if err != nil {
-		impl.logger.Debugw("Error", err)
-		return nil, err
-	}
-
-	// fetching material info separately because it gives error with fetching other (check its json) - FIXME
 	type Object struct {
 		Id           int    `json:"id"`
 		MaterialInfo string `json:"material_info"`
 	}
-
 	var artifactsB []Object
-	var queryTemp string = "SELECT cia.id, cia.material_info FROM ci_artifact cia" +
-		" INNER JOIN ci_pipeline cp on cp.id=cia.pipeline_id" +
-		" INNER JOIN pipeline p on p.ci_pipeline_id = cp.id" +
-		" WHERE p.id= ? ORDER BY cia.id DESC"
-	_, err = impl.dbConnection.Query(&artifactsB, queryTemp, cdPipelineId)
-	if err != nil {
-		return nil, err
+
+	if parentType == bean.WEBHOOK_WORKFLOW_TYPE {
```

Comment:
too much logic in repository

Reply:
noted, we will re visit this section, it's a tested code so don't want much changes in this section, will do refactor this as a new activity.